pkg/gitreceive/build.go

```diff
@@ -169,51 +125,77 @@ func build(conf *Config, s3Client *s3.S3, builderKey, rawGitSha string) error {
 		return fmt.Errorf("uploading %s to %s/%s (%v)", absAppTgz, bucketName, slugBuilderInfo.TarKey(), err)
 	}
 
+	creds := storage.CredsOK()
+
+	var pod *api.Pod
+	var buildPodName string
+	if usingDockerfile {
+		buildPodName = dockerBuilderPodName(appName, gitSha.Short())
+		pod = dockerBuilderPod(
+			conf.Debug,
+			creds,
+			buildPodName,
+			conf.PodNamespace,
+			appConf.Values,
+			slugBuilderInfo.TarURL(),
+			imageName,
+		)
+	} else {
+		buildPodName = slugBuilderPodName(appName, gitSha.Short())
+		pod = slugbuilderPod(
+			conf.Debug,
+			creds,
+			buildPodName,
+			conf.PodNamespace,
+			appConf.Values,
+			slugBuilderInfo.TarURL(),
+			slugBuilderInfo.PushURL(),
+			buildPackURL,
+		)
+	}
+
 	log.Info("Starting build... but first, coffee!")
 	log.Debug("Starting pod %s", buildPodName)
-	kCreateCmd := exec.Command(
-		"kubectl",
-		fmt.Sprintf("--namespace=%s", conf.PodNamespace),
-		"create",
-		"-f",
-		finalManifestFileLocation,
-	)
-	if log.IsDebugging {
-		kCreateCmd.Stdout = os.Stdout
-	}
-	kCreateCmd.Stderr = os.Stderr
-	if err := run(kCreateCmd); err != nil {
+	log.Debug("Pod spec: %v", prettyPrintJSON(pod))
+
+	podsInterface := kubeClient.Pods(conf.PodNamespace)
+
+	newPod, err := podsInterface.Create(pod)
+	if err != nil {
 		return fmt.Errorf("creating builder pod (%s)", err)
 	}
 
-	// poll kubectl every 100ms to determine when the build pod is running
-	// TODO: use the k8s client and watch the event stream instead (https://github.com/deis/builder/issues/65)
-	for {
-		cmd := kGetCmd(conf.PodNamespace, buildPodName)
-		var out bytes.Buffer
-		cmd.Stdout = &out
-		// ignore errors
-		run(cmd)
-		outStr := string(out.Bytes())
-		if strings.Contains(outStr, "phase: Running") {
+	watcher, err := podsInterface.Watch(labels.Nothing(), fields.OneTermEqualSelector("name", pod.Name), v1Version)
```

Comment:
Good point. I added resourceVersion to indicate what version should be used to begin watching